### DISCUSSION TITLE
Is there a way to "manipulate" a command argument (ctx)

### DISCUSSION THREAD

**Discussion by EpicBirb** [nextcord/nextcord#447] [Q&A] [answered]
If I have a client command like `ping`, it would look like this:
```py
@client.command()
async def ping(ctx):
    await ctx.send(f"Pong: {round(client.latency * 1000)}ms")
```
It's an example above ^

But is there a way to "recreate" the `ctx` argument?

**Comment by TAG-Epic** [comment]
Sorry I dont understand. Could you explain more?

**Reply by EpicBirb** [reply]
the example above, ctx can be use as the following

```py
ctx.send
ctx.message.id
```

When the function is called when the command is executed, I think it passes something through ctx (i think it was a nextcord object)? Is there a way to recreate it (ctx)?

**Comment by DenverCoder1** [comment, marked as answer]
Short answer:

In most cases you don't need to create your own Context. If you want to use the command's logic outside the command, you can separate the logic into a different function that does not require ctx and use that when calling the method and when not.

Long answer:

The class of ctx is [`nextcord.ext.commands.Context`](https://github.com/nextcord/nextcord/blob/master/nextcord/ext/commands/context.py#L69).

If you have a message, you can do `await bot.get_context(message)` to get it.

https://nextcord.readthedocs.io/en/latest/ext/commands/api.html#nextcord.ext.commands.Bot.get_context

To fake a context, you can create a Nextcord Object and add attributes to it
```py
fake_context = nextcord.Object(id=0) # any id can be used
fake_context.bot = bot
fake_context.guild = guild
```

**Reply by EpicBirb** [reply]
tbh, I don't think I find that working. Since im using that for ctx.voice_client.play(), it's not really executing properly.

```py
ctx.voice_client.play(player, after=lambda e: print(f'Player encuntered an error: {e}') if e else asyncio.run_coroutine_threadsafe(yt.queue(asyncio.run_coroutine_threadsafe(client.get_context(context), client.loop), client, "play"), client.loop))
```
^ this is in a class

In this line we cant really use await or it will raise an error. so we have to use asyncio.run_coroutine_threadsafe(). I also have another line that will send a message using `nextcord.ext.commands.Context` object since the object has a .send() function:

```py
context = await ctx.send(f'Now playing: {player.title}')
```
But it won't send the message

**Reply by DenverCoder1** [reply]
@EpicBirb Instead of ctx, you could use fields of existing objects.

For a voice_client, possibly get/fetch a Guild object and use [Guild.voice_client](https://nextcord.readthedocs.io/en/latest/api.html#nextcord.Guild.voice_client) or [Client.voice_clients](https://nextcord.readthedocs.io/en/latest/api.html#nextcord.Guild.voice_client).

To get a voice state of a member, you can fetch the member and use Member.voice.

To send a message in a channel, you can get the channel and use [TextChannel.send](https://nextcord.readthedocs.io/en/latest/api.html#nextcord.TextChannel.send)

There are often several ways to do things, I'd recommend looking through the docs to see what's possible.

**Reply by EpicBirb** [reply]
oh ok, thanks.
